pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
+                self._device_flag = IPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int):
+                self._parallel_devices = list(range(self._device_flag))
+
+        elif self._accelerator_flag == "gpu":
+            self.accelerator = GPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = GPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int) or isinstance(self._device_flag, str):
+                self._device_flag = int(self._device_flag)
+                self._parallel_devices = (
+                    [torch.device("cuda", i) for i in device_parser.parse_gpu_ids(self._device_flag)]
+                    if self._device_flag != 0
+                    else []
+                )
+            else:
+                self._parallel_devices = [torch.device("cuda", i) for i in self._device_flag]
+
+        elif self._accelerator_flag == "cpu":
+            self.accelerator = CPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = CPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int):
+                self._parallel_devices = [torch.device("cpu")] * self._device_flag
+            else:
+                rank_zero_warn(
                     "The flag `devices` must be an int with `accelerator='cpu'`,"
-                    f" got `devices={self.devices}` instead."
+                    f" got `devices={self._device_flag}` instead."
                 )
-            self.num_processes = self.devices
-            return True
-        return False
-
-    @property
-    def use_dp(self) -> bool:
-        return self._strategy_type == _StrategyType.DP
-
-    @property
-    def use_ddp(self) -> bool:
-        return self._strategy_type in (
-            _StrategyType.BAGUA,
-            _StrategyType.DDP,
-            _StrategyType.DDP_SPAWN,
-            _StrategyType.DDP_SHARDED,
-            _StrategyType.DDP_SHARDED_SPAWN,
-            _StrategyType.DDP_FULLY_SHARDED,
-            _StrategyType.DEEPSPEED,
-            _StrategyType.TPU_SPAWN,
-        )
 
-    @property
-    def use_ddp2(self) -> bool:
-        return self._strategy_type == _StrategyType.DDP2
-
-    @property
-    def use_horovod(self) -> bool:
-        return self._strategy_type == _StrategyType.HOROVOD
-
-    @property
-    def use_deepspeed(self) -> bool:
-        return self._strategy_type == _StrategyType.DEEPSPEED
+        self._gpus = self._device_flag if not self._gpus else self._gpus
+        self._tpu_cores = self._device_flag if not self._tpu_cores else self._tpu_cores
 
-    @property
-    def use_bagua(self) -> bool:
-        return self._strategy_type == _StrategyType.BAGUA
+    def _choose_and_init_cluster_environment(self) -> None:
+        self.cluster_environment: ClusterEnvironment = LightningEnvironment()
```

Comment:
Why instantiating. Would be cleaner to return and instantiate otherwise.

Reply:
make sense, updated.